src/CMakeLists.txt

```diff
@@ -924,7 +954,7 @@ add_custom_command(TARGET ${EXECUTABLE_NAME}
 # Build binary intended to be used by bootloader
 set(EXECUTABLE_MCUBOOT_NAME "pinetime-mcuboot-app")
 set(EXECUTABLE_MCUBOOT_FILE_NAME ${EXECUTABLE_MCUBOOT_NAME}-${pinetime_VERSION_MAJOR}.${pinetime_VERSION_MINOR}.${pinetime_VERSION_PATCH})
-set(IMAGE_MCUBOOT_FILE_NAME ${EXECUTABLE_MCUBOOT_NAME}-image-${pinetime_VERSION_MAJOR}.${pinetime_VERSION_MINOR}.${pinetime_VERSION_PATCH}.bin)
+set(IMAGE_MCUBOOT_FILE_NAME ${EXECUTABLE_MCUBOOT_NAME}-image-${pinetime_VERSION_MAJOR}.${pinetime_VERSION_MINOR}.${pinetime_VERSION_PATCH}.hex)
```

Comment:
why this change of the extension from `bin` to `hex`?

Reply:
The hex format stores the target offset as well, which enables flashing without explicitly having to specify into which memory region (usually `0x8000` for mcuboot images). This is especially useful because the nrfjprog tool does not support specifying manual offsets. 

For testing the bootloader and main firmware interaction, I often leave the bootloader on the device and only flash a new mcuboot image (via SWD), instead of erasing the entire flash memory.

In addition, tools like `mergehex` operate on the hex format as well.

Does the bin format offer any advantages? Or are there other disadvantages of the hex format in the current toolchain?

Reply:
Original source of the commit comes from at https://github.com/ildar/InfiniTime/tree/p8 at https://github.com/ildar/InfiniTime/commit/1a4798f816f9d38c4da1efb9a8e37253511fa6f5 .

Reply:
This seems like a valid reasoning, unfortunately this is not my expertise. Maybe @JF002 @Avamander @Riksu9000 have more insight to this bin to hex change